check_limits.py

```diff
@@ -1,17 +1,23 @@
 
-def battery_is_ok(temperature, soc, charge_rate):
-  if temperature < 0 or temperature > 45:
-    print('Temperature is out of range!')
-    return False
-  elif soc < 20 or soc > 80:
-    print('State of Charge is out of range!')
-    return False
-  elif charge_rate > 0.8:
-    print('Charge rate is out of range!')
-    return False
 
+def check_temp_limits(temperature):
+     if temperature < 0 or temperature > 45:
+        return False
+     return True
+     
+def check_soc_limits(soc):
+  if soc < 20 or soc > 80:
+      return False
   return True
-
+  
+def check_charge_rate (charge_rate):
+  if charge_rate > 0.8:
+      return False
+  return True
+  
+def battery_is_ok(temperature, soc, charge_rate):
+  return check_temp_limits(temperature) and check_soc_limits(soc) and check_charge_rate(charge_rate)
```

Comment:
what about printing error message ? requirements appear to have been weakly implemented following refactoring.